Dependency on org.junit in non-test code

# Version

5.1.0


# Description

Happens when looking at vutuv/lib/vutuv/browserstack.ex from vutuv/vutuv here on github in !!WebStorm!! with this plugin enabled.


# Exception


## Message

org/junit/ComparisonFailure


## Stacktrace

```
java.lang.NoClassDefFoundError: org/junit/ComparisonFailure
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.quoteLiteral(ElixirPsiImplUtil.java:4714)
	at org.elixir_lang.psi.impl.ElixirStringLineImpl.quoteLiteral(ElixirStringLineImpl.java:90)
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.quotedChildNodes(ElixirPsiImplUtil.java:5772)
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.quotedChildNodes(ElixirPsiImplUtil.java:5730)
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.quote(ElixirPsiImplUtil.java:3530)
	at org.elixir_lang.psi.impl.ElixirStringLineImpl.quote(ElixirStringLineImpl.java:70)
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.quoteAsAtom(ElixirPsiImplUtil.java:3301)
	at org.elixir_lang.psi.impl.ElixirStringLineImpl.quoteAsAtom(ElixirStringLineImpl.java:75)
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.quote(ElixirPsiImplUtil.java:3690)
	at org.elixir_lang.psi.impl.ElixirKeywordKeyImpl.quote(ElixirKeywordKeyImpl.java:56)
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.hasKeywordKey(ElixirPsiImplUtil.java:3107)
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.keywordValue(ElixirPsiImplUtil.java:778)
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.keywordArgument(ElixirPsiImplUtil.java:750)
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.hasDoBlockOrKeyword(ElixirPsiImplUtil.java:3084)
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.useScopeSelector(ElixirPsiImplUtil.java:586)
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.createsNewScope(ElixirPsiImplUtil.java:599)
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.processDeclarationsInPreviousSibling(ElixirPsiImplUtil.java:1705)
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.processDeclarations(ElixirPsiImplUtil.java:1605)
	at org.elixir_lang.psi.impl.ElixirStabBodyImpl.processDeclarations(ElixirStabBodyImpl.java:57)
	at com.intellij.psi.util.PsiTreeUtil.treeWalkUp(PsiTreeUtil.java:1075)
	at org.elixir_lang.psi.scope.variable.MultiResolve.resolveResultList(MultiResolve.java:90)
	at org.elixir_lang.psi.scope.variable.MultiResolve.resolveResultList(MultiResolve.java:68)
	at org.elixir_lang.reference.Callable.multiResolve(Callable.java:693)
	at org.elixir_lang.annonator.Callable$1.visitNonModuleAttributeCall(Callable.java:103)
	at org.elixir_lang.annonator.Callable$1.visitCall(Callable.java:68)
	at org.elixir_lang.annonator.Callable$1.visitElement(Callable.java:57)
	at com.intellij.psi.impl.PsiElementBase.accept(PsiElementBase.java:274)
	at org.elixir_lang.psi.impl.ElixirUnmatchedUnqualifiedNoArgumentsCallImpl.accept(ElixirUnmatchedUnqualifiedNoArgumentsCallImpl.java:42)
	at org.elixir_lang.annonator.Callable.annotate(Callable.java:48)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.a(DefaultHighlightVisitor.java:139)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.visit(DefaultHighlightVisitor.java:102)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.a(GeneralHighlightingPass.java:368)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.a(GeneralHighlightingPass.java:307)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.a(GeneralHighlightingPass.java:327)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.a(GeneralHighlightingPass.java:330)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.analyze(DefaultHighlightVisitor.java:86)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.a(GeneralHighlightingPass.java:330)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.a(GeneralHighlightingPass.java:298)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectInformationWithProgress(GeneralHighlightingPass.java:238)
	at com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:83)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:70)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.b(PassExecutorService.java:438)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1153)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.d(PassExecutorService.java:431)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:556)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:501)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:66)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.c(PassExecutorService.java:430)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.a(PassExecutorService.java:406)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:142)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:242)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:404)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:165)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.lang.ClassNotFoundException: org.junit.ComparisonFailure PluginClassLoader[org.elixir_lang, 5.1.0]
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:65)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 57 more

```

--- a/src/org/elixir_lang/psi/impl/ElixirPsiImplUtil.java
+++ b/src/org/elixir_lang/psi/impl/ElixirPsiImplUtil.java
@@ -59,12 +59,12 @@ import org.jetbrains.annotations.Nullable;
 
 import javax.swing.*;
 import java.math.BigInteger;
+import java.nio.charset.Charset;
 import java.util.*;
 import java.util.function.Predicate;
 import java.util.stream.Stream;
 
 import static org.elixir_lang.errorreport.Logger.error;
-import static org.elixir_lang.intellij_elixir.Quoter.*;
 import static org.elixir_lang.psi.call.name.Function.*;
 import static org.elixir_lang.psi.call.name.Module.*;
 import static org.elixir_lang.psi.stub.type.call.Stub.isModular;
@@ -254,6 +254,114 @@ public class ElixirPsiImplUtil {
     @NotNull
     public static int base(@NotNull @SuppressWarnings("unused") final ElixirUnknownBaseWholeNumber unknownBaseWholeNumber) {
         return UNKNOWN_BASE;
+    }
+
+    @NotNull
+    private static OtpErlangObject elixirCharList(@NotNull final List<Integer> codePointList) {
+        OtpErlangList elixirCodePointList = elixirCodePointList(codePointList);
+
+        return elixirCharList(elixirCodePointList);
+    }
+
+    /**
+     * Erlang will automatically stringify a list that is just a list of LATIN-1 printable code
+     * points.
+     * OtpErlangString and OtpErlangList are not equal when they have the same content, so to check against
+     * Elixir.Code.string_to_quoted, this code must determine if Erlang would return an OtpErlangString instead
+     * of OtpErlangList and do the same.
+     */
+    @NotNull
+    private static OtpErlangObject elixirCharList(@NotNull final OtpErlangList erlangList) {
+        OtpErlangObject charList;
+
+        /* JInterface will return an OtpErlangString in some case and an OtpErlangList in other.  Right now, I'm
+           assuming it works similar to the printing in `iex` and is based on whether the codePoint is printable, but
+           ASCII printable instead of Unicode printable since Erlang is ASCII/LATIN-1 based */
+        if (isErlangPrintable(erlangList))  {
+            try {
+                charList = new OtpErlangString(erlangList);
+            } catch (OtpErlangException e) {
+                throw new NotImplementedException(e);
+            }
+        } else {
+            charList = erlangList;
+        }
+
+        return charList;
+    }
+
+    @NotNull
+    private static OtpErlangList elixirCodePointList(@NotNull final List<Integer> codePointList) {
+        OtpErlangLong[] erlangCodePoints = new OtpErlangLong[codePointList.size()];
+
+        int i = 0;
+        for (int codePoint : codePointList) {
+            erlangCodePoints[i++] = new OtpErlangLong(codePoint);
+        }
+
+        return new OtpErlangList(erlangCodePoints);
+    }
+
+    @NotNull
+    private static OtpErlangBinary elixirString(@NotNull final List<Integer> codePointList) {
+        StringBuilder stringAccumulator = new StringBuilder();
+
+        for (int codePoint : codePointList) {
+            stringAccumulator.appendCodePoint(codePoint);
+        }
+
+        return elixirString(stringAccumulator.toString());
+    }
+
+    @NotNull
+    public static OtpErlangBinary elixirString(@NotNull String javaString) {
+        final byte[] bytes = javaString.getBytes(Charset.forName("UTF-8"));
+        return new OtpErlangBinary(bytes);
+    }
+
+    private static boolean isErlangPrintable(@NotNull final OtpErlangList erlangList) {
+        boolean isErlangPrintable = true;
+
+        for (OtpErlangObject erlangObject : erlangList) {
+
+            if (erlangObject instanceof OtpErlangLong) {
+                OtpErlangLong erlangLong = (OtpErlangLong) erlangObject;
+
+                final int codePoint;
+
+                try {
+                    codePoint = erlangLong.intValue();
+                } catch (OtpErlangRangeException e) {
+                    isErlangPrintable = false;
+                    break;
+                }
+
+                if (!isErlangPrintable(codePoint)) {
+                    isErlangPrintable = false;
+                    break;
+                }
+            } else {
+                isErlangPrintable = false;
+                break;
+            }
+        }
+
+        if (erlangList.arity() == 0) {
+            isErlangPrintable = false;
+        }
+
+        return isErlangPrintable;
+    }
+
+    @Contract(pure = true)
+    private static boolean isErlangPrintable(int codePoint) {
+        return (codePoint >= 0 && codePoint <= 255);
+    }
+
+    @NotNull
+    public static String javaString(@NotNull OtpErlangBinary elixirString) {
+        byte[] bytes = elixirString.binaryValue();
+        return new String(bytes, Charset.forName("UTF-8"));
     }
 
     /**

--- a/tests/org/elixir_lang/intellij_elixir/Quoter.java
+++ b/tests/org/elixir_lang/intellij_elixir/Quoter.java
@@ -24,14 +24,6 @@ public class Quoter {
     private static final String REMOTE_NAME = "Elixir.IntellijElixir.Quoter";
     private static final int TIMEOUT_IN_MILLISECONDS = 1000;
 
-    @Contract("null -> fail")
-    private static void assertMessageReceived(OtpErlangObject message) {
-        assertNotNull(
-                "did not receive message from " + REMOTE_NAME + "@" + IntellijElixir.REMOTE_NODE + ".  Make sure it is running",
-                message
-        );
-    }
-
     public static void assertError(PsiFile file) {
         final String text = file.getText();
 
@@ -43,8 +35,7 @@ public class Quoter {
             String statusString = status.atomValue();
 
             assertEquals(statusString, "error");
-        }
-        catch (IOException | OtpErlangDecodeException | OtpErlangExit e) {
+        } catch (IOException | OtpErlangDecodeException | OtpErlangExit e) {
             throw new RuntimeException(e);
         }
     }
@@ -55,12 +46,19 @@ public class Quoter {
 
         try {
             Quoter.quote(text);
-        }
-        catch (IOException | OtpErlangDecodeException | OtpErlangExit e) {
+        } catch (IOException | OtpErlangDecodeException | OtpErlangExit e) {
             exception = e;
         }
 
         assertThat(exception, instanceOf(OtpErlangExit.class));
+    }
+
+    @Contract("null -> fail")
+    private static void assertMessageReceived(OtpErlangObject message) {
+        assertNotNull(
+                "did not receive message from " + REMOTE_NAME + "@" + IntellijElixir.REMOTE_NODE + ".  Make sure it is running",
+                message
+        );
     }
 
     public static void assertQuotedCorrectly(PsiFile file) {
@@ -89,20 +87,19 @@ public class Quoter {
                 String token = ElixirPsiImplUtil.javaString(tokenBinary);
 
                 throw new AssertionError(
-                        "intellij_elixir returned \"" + message + "\" on line " + line + " due to " + token  +
+                        "intellij_elixir returned \"" + message + "\" on line " + line + " due to " + token +
                                 ", use assertQuotesAroundError if error is expect in Elixir natively, " +
                                 "but not in intellij-elixir plugin"
                 );
             }
-        }
-        catch (IOException | OtpErlangDecodeException | OtpErlangExit e) {
+        } catch (IOException | OtpErlangDecodeException | OtpErlangExit e) {
             throw new RuntimeException(e);
         }
     }
 
     private static void assertQuotedCorrectly(@NotNull OtpErlangObject expectedQuoted,
                                               @NotNull OtpErlangObject actualQuoted) {
-        if(!expectedQuoted.equals(actualQuoted)) {
+        if (!expectedQuoted.equals(actualQuoted)) {
             throw new ComparisonFailure(null, toString(expectedQuoted), toString(actualQuoted));
         }
     }
@@ -148,7 +145,7 @@ public class Quoter {
 
         stringBuilder.append("[");
 
-        for (int i = 0; i< quoted.arity(); i++) {
+        for (int i = 0; i < quoted.arity(); i++) {
             if (i > 0) {
                 stringBuilder.append(",");
             }

--- a/tests/org/elixir_lang/intellij_elixir/Quoter.java
+++ b/tests/org/elixir_lang/intellij_elixir/Quoter.java
@@ -2,7 +2,6 @@ package org.elixir_lang.intellij_elixir;
 
 import com.ericsson.otp.erlang.*;
 import com.intellij.psi.PsiFile;
-import org.apache.commons.lang.NotImplementedException;
 import org.elixir_lang.GenericServer;
 import org.elixir_lang.IntellijElixir;
 import org.elixir_lang.psi.impl.ElixirPsiImplUtil;
@@ -12,7 +11,6 @@ import org.junit.ComparisonFailure;
 
 import java.io.IOException;
 import java.nio.charset.Charset;
-import java.util.List;
 
 import static org.apache.commons.lang.CharUtils.isAsciiPrintable;
 import static org.hamcrest.CoreMatchers.instanceOf;
@@ -95,10 +93,10 @@ public class Quoter {
                 OtpErlangLong line = (OtpErlangLong) error.elementAt(0);
 
                 OtpErlangBinary messageBinary = (OtpErlangBinary) error.elementAt(1);
-                String message = javaString(messageBinary);
+                String message = ElixirPsiImplUtil.javaString(messageBinary);
 
                 OtpErlangBinary tokenBinary = (OtpErlangBinary) error.elementAt(2);
-                String token = javaString(tokenBinary);
+                String token = ElixirPsiImplUtil.javaString(tokenBinary);
 
                 throw new AssertionError(
                         "intellij_elixir returned \"" + message + "\" on line " + line + " due to " + token  +
@@ -132,118 +130,10 @@ public class Quoter {
         return new String(actualBytes, Charset.forName("UTF-8"));
     }
 
-    @NotNull
-    public static OtpErlangObject elixirCharList(@NotNull final List<Integer> codePointList) {
-        OtpErlangList elixirCodePointList = elixirCodePointList(codePointList);
-
-        return elixirCharList(elixirCodePointList);
-    }
-
-    /*
-     * Erlang will automatically stringify a list that is just a list of LATIN-1 printable code
-     * points.
-     * OtpErlangString and OtpErlangList are not equal when they have the same content, so to check against
-     * Elixir.Code.string_to_quoted, this code must determine if Erlang would return an OtpErlangString instead
-     * of OtpErlangList and do the same.
-     */
-    @NotNull
-    public static OtpErlangObject elixirCharList(@NotNull final OtpErlangList erlangList) {
-        OtpErlangObject charList;
-
-        /* JInterface will return an OtpErlangString in some case and an OtpErlangList in other.  Right now, I'm
-           assuming it works similar to the printing in `iex` and is based on whether the codePoint is printable, but
-           ASCII printable instead of Unicode printable since Erlang is ASCII/LATIN-1 based */
-        if (isErlangPrintable(erlangList))  {
-            try {
-                charList = new OtpErlangString(erlangList);
-            } catch (OtpErlangException e) {
-                throw new NotImplementedException(e);
-            }
-        } else {
-            charList = erlangList;
-        }
-
-        return charList;
-    }
-
-    @NotNull
-    public static OtpErlangList elixirCodePointList(@NotNull final List<Integer> codePointList) {
-        OtpErlangLong[] erlangCodePoints = new OtpErlangLong[codePointList.size()];
-
-        int i = 0;
-        for (int codePoint : codePointList) {
-            erlangCodePoints[i++] = new OtpErlangLong(codePoint);
-        }
-
-        return new OtpErlangList(erlangCodePoints);
-    }
-
-    public static boolean isErlangPrintable(@NotNull final OtpErlangList erlangList) {
-        boolean isErlangPrintable = true;
-
-        for (OtpErlangObject erlangObject : erlangList) {
-
-            if (erlangObject instanceof OtpErlangLong) {
-                OtpErlangLong erlangLong = (OtpErlangLong) erlangObject;
-
-                final int codePoint;
-
-                try {
-                    codePoint = erlangLong.intValue();
-                } catch (OtpErlangRangeException e) {
-                    isErlangPrintable = false;
-                    break;
-                }
-
-                if (!isErlangPrintable(codePoint)) {
-                    isErlangPrintable = false;
-                    break;
-                }
-            } else {
-                isErlangPrintable = false;
-                break;
-            }
-        }
-
-        if (erlangList.arity() == 0) {
-            isErlangPrintable = false;
-        }
-
-        return isErlangPrintable;
-    }
-
-    @NotNull
-    public static OtpErlangBinary elixirString(@NotNull final List<Integer> codePointList) {
-        StringBuilder stringAccumulator = new StringBuilder();
-
-        for (int codePoint : codePointList) {
-            stringAccumulator.appendCodePoint(codePoint);
-        }
-
-        return elixirString(stringAccumulator.toString());
-    }
-
-    @NotNull
-    public static OtpErlangBinary elixirString(@NotNull String javaString) {
-        final byte[] bytes = javaString.getBytes(Charset.forName("UTF-8"));
-        return new OtpErlangBinary(bytes);
-    }
-
-    @Contract(pure = true)
-    public static boolean isErlangPrintable(int codePoint) {
-        return (codePoint >= 0 && codePoint <= 255);
-    }
-
-    @NotNull
-    public static String javaString(@NotNull OtpErlangBinary elixirString) {
-        byte[] bytes = elixirString.binaryValue();
-        return new String(bytes, Charset.forName("UTF-8"));
-    }
-
     public static OtpErlangTuple quote(@NotNull String code) throws IOException, OtpErlangExit, OtpErlangDecodeException {
         final OtpNode otpNode = IntellijElixir.getLocalNode();
         final OtpMbox otpMbox = otpNode.createMbox();
-        OtpErlangObject request = elixirString(code);
+        OtpErlangObject request = ElixirPsiImplUtil.elixirString(code);
 
         return (OtpErlangTuple) GenericServer.call(
                 otpMbox,

--- a/tests/org/elixir_lang/intellij_elixir/Quoter.java
+++ b/tests/org/elixir_lang/intellij_elixir/Quoter.java
@@ -10,7 +10,6 @@ import org.jetbrains.annotations.NotNull;
 import org.junit.ComparisonFailure;
 
 import java.io.IOException;
-import java.nio.charset.Charset;
 
 import static org.apache.commons.lang.CharUtils.isAsciiPrintable;
 import static org.hamcrest.CoreMatchers.instanceOf;
@@ -20,14 +19,13 @@ import static org.junit.Assert.*;
  * Created by luke.imhoff on 12/31/14.
  */
 public class Quoter {
-    public static final OtpErlangAtom CODE_KEY = new OtpErlangAtom("code");
-    public static final OtpErlangAtom QUOTED_KEY = new OtpErlangAtom("quoted");
     /* remote name is Elixir.IntellijElixir.Quoter because all aliases in Elixir look like atoms prefixed with
        with Elixir. from erlang's perspective. */
-    public static final String REMOTE_NAME = "Elixir.IntellijElixir.Quoter";
-    public static final int TIMEOUT_IN_MILLISECONDS = 1000;
+    private static final String REMOTE_NAME = "Elixir.IntellijElixir.Quoter";
+    private static final int TIMEOUT_IN_MILLISECONDS = 1000;
 
-    public static void assertMessageReceived(OtpErlangObject message) {
+    @Contract("null -> fail")
+    private static void assertMessageReceived(OtpErlangObject message) {
         assertNotNull(
                 "did not receive message from " + REMOTE_NAME + "@" + IntellijElixir.REMOTE_NODE + ".  Make sure it is running",
                 message
@@ -46,11 +44,7 @@ public class Quoter {
 
             assertEquals(statusString, "error");
         }
-        catch (IOException e) {
-            throw new RuntimeException(e);
-        } catch (OtpErlangDecodeException e) {
-            throw new RuntimeException(e);
-        } catch (OtpErlangExit e) {
+        catch (IOException | OtpErlangDecodeException | OtpErlangExit e) {
             throw new RuntimeException(e);
         }
     }
@@ -62,12 +56,8 @@ public class Quoter {
         try {
             Quoter.quote(text);
         }
-        catch (IOException ioExeption) {
-            exception = ioExeption;
-        } catch (OtpErlangDecodeException otpErlangDecodeException) {
-            exception = otpErlangDecodeException;
-        } catch (OtpErlangExit otpErlangExit) {
-            exception = otpErlangExit;
+        catch (IOException | OtpErlangDecodeException | OtpErlangExit e) {
+            exception = e;
         }
 
         assertThat(exception, instanceOf(OtpErlangExit.class));
@@ -105,29 +95,16 @@ public class Quoter {
                 );
             }
         }
-        catch (IOException e) {
-            throw new RuntimeException(e);
-        } catch (OtpErlangDecodeException e) {
-            throw new RuntimeException(e);
-        } catch (OtpErlangExit e) {
+        catch (IOException | OtpErlangDecodeException | OtpErlangExit e) {
             throw new RuntimeException(e);
         }
     }
 
-    public static void assertQuotedCorrectly(@NotNull OtpErlangObject expectedQuoted,
-                                             @NotNull OtpErlangObject actualQuoted) {
+    private static void assertQuotedCorrectly(@NotNull OtpErlangObject expectedQuoted,
+                                              @NotNull OtpErlangObject actualQuoted) {
         if(!expectedQuoted.equals(actualQuoted)) {
             throw new ComparisonFailure(null, toString(expectedQuoted), toString(actualQuoted));
         }
-    }
-
-    @Contract(pure = true)
-    @NotNull
-    public static String code(@NotNull OtpErlangMap quotedMessageMap) {
-        OtpErlangBinary actualElixirString = (OtpErlangBinary) quotedMessageMap.get(CODE_KEY);
-        byte[] actualBytes = actualElixirString.binaryValue();
-
-        return new String(actualBytes, Charset.forName("UTF-8"));
     }
 
     public static OtpErlangTuple quote(@NotNull String code) throws IOException, OtpErlangExit, OtpErlangDecodeException {
@@ -146,7 +123,7 @@ public class Quoter {
     }
 
     @NotNull
-    public static String toString(@NotNull OtpErlangBitstr quoted) {
+    private static String toString(@NotNull OtpErlangBitstr quoted) {
         StringBuilder stringBuilder = new StringBuilder();
         stringBuilder.append('"');
 
@@ -166,7 +143,7 @@ public class Quoter {
     }
 
     @NotNull
-    public static String toString(@NotNull OtpErlangList quoted) {
+    private static String toString(@NotNull OtpErlangList quoted) {
         StringBuilder stringBuilder = new StringBuilder();
 
         stringBuilder.append("[");
@@ -185,7 +162,7 @@ public class Quoter {
     }
 
     @NotNull
-    public static String toString(@NotNull OtpErlangObject quoted) {
+    private static String toString(@NotNull OtpErlangObject quoted) {
         String string;
 
         if (quoted instanceof OtpErlangBoolean ||
@@ -216,7 +193,7 @@ public class Quoter {
     }
 
     @NotNull
-    public static String toString(@NotNull OtpErlangTuple quoted) {
+    private static String toString(@NotNull OtpErlangTuple quoted) {
         StringBuilder stringBuilder = new StringBuilder();
 
         stringBuilder.append("{");

